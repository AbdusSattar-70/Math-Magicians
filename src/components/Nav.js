import React from 'react';
import { NavLink } from 'react-router-dom';

import styles from '../styles/Nav.module.css';

function Nav() {
  return (
    <header>
      <nav>
        <NavLink to="/" className={styles.navLink}>
          Home
        </NavLink>
        <NavLink to="/calculator" className={styles.navLink}>
          Calculator
        </NavLink>
        <NavLink to="/quotes" className={styles.navLink}>
          Quotes
        </NavLink>
      </nav>
    </header>
  );
}

export default Nav;
