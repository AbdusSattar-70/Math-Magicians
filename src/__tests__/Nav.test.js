import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Nav from '../components/Nav';

it('test Header rendering', () => {
  render(
    <BrowserRouter>
      <Nav />
    </BrowserRouter>,
  );
  const H1 = screen.getByText(/Math Magicians/i);
  expect(H1.textContent).toBe('Math Magicians');
});

it('test navbar render', () => {
  render(
    <BrowserRouter>
      <Nav />
    </BrowserRouter>,
  );
  const tree = renderer
    .create(
      <BrowserRouter>
        <Nav />
      </BrowserRouter>,
    )
    .toJSON();
  const navbar = screen.getAllByRole('link');
  expect(navbar).toHaveLength(3);
  expect(tree).toMatchSnapshot();
});
