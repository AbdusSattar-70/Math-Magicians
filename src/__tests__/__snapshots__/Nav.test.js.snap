// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test navbar render 1`] = `
<header
  className="header"
>
  <h1>
    Math Magicians
  </h1>
  <nav>
    <ul>
      <li>
        <a
          aria-current="page"
          className="active"
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <hr />
      <li>
        <a
          className=""
          href="/calculator"
          onClick={[Function]}
        >
          Calculator
        </a>
      </li>
      <hr />
      <li>
        <a
          className=""
          href="/quotes"
          onClick={[Function]}
        >
          Quotes
        </a>
      </li>
    </ul>
  </nav>
</header>
`;
